# HG changeset patch
# User Rik <rik@octave.org>
# Date 1547012153 28800
#      Tue Jan 08 21:35:53 2019 -0800
# Node ID 7da6136d3ecde8cc1b524fe44839db6b89b97be8
# Parent  10919e3c9aaaa42a3cc92c15d1a18084b5c127ac
# Parent  0fba2d918b6543a66da074d59f0b62f147492ed3
maint: merge stable to default.

diff --git a/libgui/languages/pt_BR.ts b/libgui/languages/pt_BR.ts
--- a/libgui/languages/pt_BR.ts
+++ b/libgui/languages/pt_BR.ts
@@ -97,17 +97,17 @@
     <message>
         <location line="+5"/>
         <source>Select All</source>
         <translation>Selecionar Tudo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Run Selection</source>
-        <translation type="unfinished">Executar Seleção</translation>
+        <translation>Executar Seleção</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Clear Window</source>
         <translation>Limpar Tudo</translation>
     </message>
     <message>
         <location filename="../qterminal/libqterminal/QTerminal.cc" line="+49"/>
@@ -383,74 +383,76 @@ file. Search capabilities may be affecte
 temporário. A funcionalidade de busca pode estar afetada.
 %1</translation>
     </message>
     <message>
         <location line="+12"/>
         <source>Could not setup the data required for the
 documentation viewer. Only help texts in
 the Command Window will be available.</source>
-        <translation type="unfinished"></translation>
+        <translation>Não foi possível configurar os dados necessários
+para o visualizador de documentação. Apenas textos
+de ajuda na Janela de Comando estarão disponíveis.</translation>
     </message>
     <message>
         <location line="+21"/>
         <source>Find:</source>
-        <translation>Buscar:</translation>
+        <translation>Localizar:</translation>
     </message>
     <message>
         <location line="+7"/>
         <location line="+1"/>
         <source>Search forward</source>
         <translation>Buscar adiante</translation>
     </message>
     <message>
         <location line="+5"/>
         <location line="+1"/>
         <source>Search backward</source>
         <translation>Buscar para trás</translation>
     </message>
     <message>
         <location line="+74"/>
         <source>Function Index</source>
-        <translation type="unfinished"></translation>
+        <translation>Índice de Funções</translation>
     </message>
     <message>
         <location line="+98"/>
         <source>Go home</source>
-        <translation type="unfinished"></translation>
+        <translation>Ir para o início</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Go back</source>
-        <translation type="unfinished"></translation>
+        <translation>Ir para trás</translation>
     </message>
     <message>
         <location line="+14"/>
         <source>Go forward</source>
-        <translation type="unfinished"></translation>
+        <translation>Ir para frente</translation>
     </message>
     <message>
         <location line="+43"/>
         <source>Find</source>
-        <translation type="unfinished">Localizar</translation>
+        <translation>Localizar</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Zoom in</source>
-        <translation type="unfinished"></translation>
+        <translation>Mais zoom</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Zoom out</source>
-        <translation type="unfinished"></translation>
+        <translation>Menos zoom</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Zoom original</source>
-        <translation type="unfinished"></translation>
+        <translation>Zoom original</translation>
     </message>
     <message>
         <location line="-206"/>
         <source>Contents</source>
         <translation>Conteúdo</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -461,22 +463,22 @@ the Command Window will be available.</s
         <location line="+10"/>
         <location line="+39"/>
         <source>Search</source>
         <translation>Buscar</translation>
     </message>
     <message>
         <location line="+84"/>
         <source>Previous pages</source>
-        <translation type="unfinished"></translation>
+        <translation>Páginas anteriores</translation>
     </message>
     <message>
         <location line="+14"/>
         <source>Next pages</source>
-        <translation type="unfinished"></translation>
+        <translation>Páginas seguintes</translation>
     </message>
     <message>
         <location line="+383"/>
         <source>Unable to register help file %1.</source>
         <translation>Não foi possível registrar o arquivo de ajuda %1.</translation>
     </message>
 </context>
 <context>
@@ -492,37 +494,37 @@ the Command Window will be available.</s
         <translation>Veja a documentação para obter ajuda.</translation>
     </message>
 </context>
 <context>
     <name>octave::dw_main_window</name>
     <message>
         <location filename="../src/dw-main-window.cc" line="+49"/>
         <source>&amp;Close</source>
-        <translation type="unfinished">&amp;Fechar</translation>
+        <translation>&amp;Fechar</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Close &amp;All</source>
-        <translation type="unfinished"></translation>
+        <translation>Fechar &amp;Todos</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Close &amp;Other</source>
-        <translation type="unfinished"></translation>
+        <translation>Fechar as &amp;Demais Abas</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Switch to &amp;Left Widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Alternar para Aba à &amp;Esquerda</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Switch to &amp;Right Widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Alternar para Aba à &amp;Direita</translation>
     </message>
 </context>
 <context>
     <name>octave::external_editor_interface</name>
     <message>
         <location filename="../src/external-editor-interface.cc" line="+63"/>
         <location line="+45"/>
         <source>Octave Editor</source>
@@ -797,17 +799,17 @@ para escrita: %2.</translation>
     <message>
         <location line="+3"/>
         <source>Navi&amp;gation</source>
         <translation>Nave&amp;gação</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Go &amp;to Line...</source>
-        <translation>Vá para &amp;Linha...</translation>
+        <translation>Ir para &amp;Linha...</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Move to Matching Brace</source>
         <translation>Mover para Parêntese/Chave/Colchete Correspondente</translation>
     </message>
     <message>
         <location line="+5"/>
@@ -1014,22 +1016,23 @@ para escrita: %2.</translation>
         <location line="+36"/>
         <source>Octave Editor</source>
         <translation>Editor do Octave</translation>
     </message>
     <message>
         <location line="-1977"/>
         <source>Cannot add breakpoint to modified or unnamed file.
 Save and add breakpoint, or cancel?</source>
-        <translation type="unfinished"></translation>
+        <translation>Não é possível adicionar o ponto de parada a arquivo modificado ou sem nome.
+Salvar e adicionar ponto de parada ou cancelar?</translation>
     </message>
     <message>
         <location line="+833"/>
         <source>Goto line</source>
-        <translation>Vá para linha</translation>
+        <translation>Ir para linha</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Line number</source>
         <translation>Número da linha</translation>
     </message>
     <message>
         <location line="+126"/>
@@ -1069,48 +1072,56 @@ está prestes a ser fechado mas foi modificado.  %2</translation>
     </message>
     <message>
         <location line="+126"/>
         <source>There were problems reading the file
 %1
 with the selected encoding %2.
 
 Modifying and saving the file might cause data loss!</source>
-        <translation type="unfinished"></translation>
+        <translation>Ocorreram problemas ao ler o arquivo
+%1
+com a codificação selecionada %2.
+
+Modificar e salvar o arquivo pode causar perda de dados!</translation>
     </message>
     <message>
         <location line="+10"/>
         <source>&amp;Edit anyway</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Editar mesmo assim</translation>
     </message>
     <message>
         <location line="+1"/>
         <location line="+57"/>
         <source>Chan&amp;ge encoding</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Mudar codificação</translation>
     </message>
     <message>
         <location line="-56"/>
         <location line="+49"/>
         <source>&amp;Close</source>
-        <translation type="unfinished">&amp;Fechar</translation>
+        <translation>&amp;Fechar</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>Select new default encoding</source>
-        <translation type="unfinished"></translation>
+        <translation>Selecionar nova codificação padrão</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Please select a new encoding
 for reloading the current file.
 
 This does not change the default encoding.
 </source>
-        <translation type="unfinished"></translation>
+        <translation>Favor selecionar uma nova codificação
+para recarregar o arquivo atual.
+
+Isso não altera a codificação padrão.
+</translation>
     </message>
     <message>
         <location line="+212"/>
         <source>Debug or Save</source>
         <translation>Depurar ou Salvar</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -1127,26 +1138,33 @@ Abandonar depuração e salvar?</translation>
 %2.</translation>
     </message>
     <message>
         <location line="+182"/>
         <source>The current encoding %1
 can not be applied.
 
 Please select another one!</source>
-        <translation type="unfinished"></translation>
+        <translation>A codificação atual %1
+não pode ser aplicada.
+
+Favor selecionar outra!</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>The current editor contents can not be encoded
 with the selected encoding %1.
 Using it would result in data loss!
 
 Please select another one!</source>
-        <translation type="unfinished"></translation>
+        <translation>O conteúdo atual do editor não pode ser codificado
+usando a codificação selecionada %1.
+Usá-la resultaria em perda de dados!
+
+Favor selecionar outra!</translation>
     </message>
     <message>
         <location line="-115"/>
         <source>Octave Files (*.m)</source>
         <translation>Arquivos do Octave (*.m)</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -1264,17 +1282,17 @@ parece ter sido deletado ou renomeado. Gostaria de salvá-lo agora?%2</translation>
     <message>
         <location line="+4"/>
         <source>Actions on current directory</source>
         <translation>Ações no diretório atual</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Show Home Directory</source>
-        <translation>Exibir Diretório Pessoal</translation>
+        <translation>Exibir Diretório Inicial</translation>
     </message>
     <message>
         <location line="+12"/>
         <source>Set Browser Directory...</source>
         <translation>Definir Diretório a Navegar...</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -1292,17 +1310,17 @@ parece ter sido deletado ou renomeado. Gostaria de salvá-lo agora?%2</translation>
         <location line="-381"/>
         <location line="+384"/>
         <source>New Directory...</source>
         <translation>Novo Diretório...</translation>
     </message>
     <message>
         <location line="-329"/>
         <source>Double-click to open file/folder, right click for alternatives</source>
-        <translation type="unfinished"></translation>
+        <translation>Clique-duplo para abrir arquivo/pasta, botão direito para alternativas</translation>
     </message>
     <message>
         <location line="+282"/>
         <source>Open</source>
         <translation>Abrir</translation>
     </message>
     <message>
         <location line="+5"/>
@@ -1785,17 +1803,17 @@ a:link { text-decoration: underline; col
     <message>
         <location line="+3"/>
         <source>Show filter</source>
         <translation>Exibir filtro</translation>
     </message>
     <message>
         <location line="+121"/>
         <source>Double-click a command to transfer it to the Command Window.</source>
-        <translation type="unfinished"></translation>
+        <translation>Dê clique-duplo sobre um comando para transferí-lo à Janela de Comandos.</translation>
     </message>
     <message>
         <location line="+8"/>
         <source>Enter text to filter the command history</source>
         <translation>Digite para filtrar o histórico de comandos</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -1922,17 +1940,18 @@ Clique em &apos;Avançar&apos; para criar um arquivo de configuração e iniciar o Octave.&lt;/p&gt;
 </source>
         <translation>Nome da nova função:
 </translation>
     </message>
     <message>
         <location line="+44"/>
         <source>%1 is a built-in, compiled or inline
 function and can not be edited.</source>
-        <translation type="unfinished"></translation>
+        <translation>%1 é uma função embutida, compilada ou inline
+e não pode ser editada.</translation>
     </message>
     <message>
         <location line="+40"/>
         <source>Can not find function %1</source>
         <translation>Não foi possível encontrar a função %1</translation>
     </message>
     <message>
         <location line="+9"/>
@@ -2419,64 +2438,64 @@ A interface gráfica do Octave precisa ser fechada agora.</translation>
         <translation>Erro Crítico no Octave</translation>
     </message>
 </context>
 <context>
     <name>octave::settings_dialog</name>
     <message>
         <location filename="../src/settings-dialog.cc" line="-573"/>
         <source>Octave Preferences</source>
-        <translation type="unfinished"></translation>
+        <translation>Preferências do Octave</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Unable to save preferences.  Missing preferences file or unknown directory.</source>
-        <translation type="unfinished"></translation>
+        <translation>Não é possível salvar as preferências. Arquivo de preferências ausente ou diretório desconhecido.</translation>
     </message>
     <message>
         <location line="+15"/>
         <location line="+4"/>
         <location line="+700"/>
         <source>System setting</source>
-        <translation type="unfinished">Configuração do sistema</translation>
+        <translation>Configuração do sistema</translation>
     </message>
     <message>
         <location line="-448"/>
         <source>IBeam Cursor</source>
-        <translation type="unfinished">Cursor IBeam</translation>
+        <translation>Cursor &quot;I&quot;</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Block Cursor</source>
-        <translation type="unfinished">Cursor de Bloco</translation>
+        <translation>Cursor de Bloco</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Underline Cursor</source>
-        <translation type="unfinished">Cursor Sublinhado</translation>
+        <translation>Cursor Sublinhado</translation>
     </message>
     <message>
         <location line="+164"/>
         <source>Set Octave Startup Directory</source>
-        <translation type="unfinished">Definir Diretório de Inicialização do Octave</translation>
+        <translation>Definir Diretório de Inicialização do Octave</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Set File Browser Startup Directory</source>
-        <translation type="unfinished">Alterar Diretório de Inicialização do Navegador de Arquivos</translation>
+        <translation>Definir Diretório de Inicialização do Navegador de Arquivos</translation>
     </message>
     <message>
         <location line="+503"/>
         <source>Enable attribute colors</source>
-        <translation type="unfinished"></translation>
+        <translation>Habilitar cores para atributos</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Hide tools tips</source>
-        <translation type="unfinished"></translation>
+        <translation>Ocultar dicas contextuais</translation>
     </message>
 </context>
 <context>
     <name>octave::setup_community_news</name>
     <message>
         <location filename="../src/welcome-wizard.cc" line="+42"/>
         <source>Community News</source>
         <translation>Novidades da Comunidade</translation>
@@ -2966,17 +2985,17 @@ Você deseja usar a tecla de atalho mesmo assim, removendo-a da outra ação?</translation>
     <message>
         <location line="+2"/>
         <source>Convert Line Endings to Mac</source>
         <translation>Converter Finais de Linha para Mac</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Goto Line</source>
-        <translation>Vá para Linha</translation>
+        <translation>Ir para Linha</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Move to Matching Brace</source>
         <translation>Mover para Parêntese/Chave/Colchete Correspondente</translation>
     </message>
     <message>
         <location line="+2"/>
@@ -3046,37 +3065,37 @@ Você deseja usar a tecla de atalho mesmo assim, removendo-a da outra ação?</translation>
     <message>
         <location line="+2"/>
         <source>Show Horizontal Scrollbar</source>
         <translation>Exibir Barra de Rolagem Horizontal</translation>
     </message>
     <message>
         <location line="+154"/>
         <source>Documentation Viewer</source>
-        <translation type="unfinished"></translation>
+        <translation>Visualizador de Documentação</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Browser</source>
-        <translation type="unfinished"></translation>
+        <translation>Navegador</translation>
     </message>
     <message>
         <location line="+79"/>
         <source>Import shortcuts from file...</source>
-        <translation type="unfinished"></translation>
+        <translation>Importar teclas de atalho do arquivo...</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Export shortcuts to file...</source>
-        <translation type="unfinished"></translation>
+        <translation>Exportar teclas de atalho para arquivo...</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>Failed to open %1 as Octave shortcut file</source>
-        <translation type="unfinished"></translation>
+        <translation>Falha ao abrir %1 como arquivo de teclas de atalho do Octave</translation>
     </message>
     <message>
         <location line="-367"/>
         <source>Zoom In</source>
         <translation>Mais Zoom</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -3127,27 +3146,27 @@ Você deseja usar a tecla de atalho mesmo assim, removendo-a da outra ação?</translation>
     <message>
         <location line="+2"/>
         <source>Document on Keyword</source>
         <translation>Documentação sobre Palavra-chave</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Go to Homepage</source>
-        <translation type="unfinished"></translation>
+        <translation>Ir para página inicial</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Go Back one Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Voltar uma Página</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Go Forward one Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Avançar uma Página</translation>
     </message>
     <message>
         <location line="+68"/>
         <source>Global</source>
         <translation>Global</translation>
     </message>
     <message>
         <location line="+3"/>
@@ -3186,22 +3205,22 @@ Você deseja usar a tecla de atalho mesmo assim, removendo-a da outra ação?</translation>
     <message>
         <location line="+2"/>
         <source>Tab Handling in Dock Widgets</source>
         <translation>Manuseio de Abas no Acoplamento de Paineis</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Find &amp; Replace in Dock Widgets</source>
-        <translation type="unfinished"></translation>
+        <translation>Localizar e Substituir em Painéis</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Zooming in Editor and Documentation</source>
-        <translation type="unfinished"></translation>
+        <translation>Zoom no Editor e na Documentação</translation>
     </message>
     <message>
         <location line="+21"/>
         <source>View Menu</source>
         <translation>Menu Exibir</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -3876,77 +3895,77 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="-962"/>
         <source>Show horizontal scroll bar</source>
         <translation>Exibir barra de rolagem horizontal</translation>
     </message>
     <message>
         <location line="-1037"/>
         <source>Preferences</source>
-        <translation type="unfinished">Preferências</translation>
+        <translation>Preferências</translation>
     </message>
     <message>
         <location line="+57"/>
         <source>(requires restart)</source>
-        <translation type="unfinished"></translation>
+        <translation>(requer reinicialização)</translation>
     </message>
     <message>
         <location line="+32"/>
         <source>Use native file dialogs</source>
-        <translation type="unfinished"></translation>
+        <translation>Usar caixas de diálogo de arquivos nativas</translation>
     </message>
     <message>
         <location line="+53"/>
         <source>Use system icon theme if available (requires restart)</source>
-        <translation type="unfinished"></translation>
+        <translation>Use o tema de ícones do sistema, se disponível (requer reinicialização)</translation>
     </message>
     <message>
         <location line="+70"/>
         <source>Toolbar Icons</source>
-        <translation type="unfinished"></translation>
+        <translation>Ícones da Barra de Ferramentas</translation>
     </message>
     <message>
         <location line="+186"/>
         <source>Language</source>
-        <translation type="unfinished"></translation>
+        <translation>Idioma</translation>
     </message>
     <message>
         <location line="+27"/>
         <source>Style</source>
-        <translation type="unfinished"></translation>
+        <translation>Estilo</translation>
     </message>
     <message>
         <location line="+27"/>
         <source>Initial working directory of Octave interpreter</source>
-        <translation type="unfinished"></translation>
+        <translation>Diretório de trabalho inicial do interpretador do Octave</translation>
     </message>
     <message>
         <location line="+18"/>
         <source>Restore last working directory of previous session</source>
-        <translation type="unfinished"></translation>
+        <translation>Restaurar último diretório de trabalho da sessão anterior</translation>
     </message>
     <message>
         <location line="+45"/>
         <source>Command</source>
-        <translation type="unfinished"></translation>
+        <translation>Comando</translation>
     </message>
     <message>
         <location line="+208"/>
         <source>Set focus to Command Window when running a command from within another widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Definir o foco na Janela de Comandos ao executar um comando de outro painel</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Print debug location in Command Window in addition to the marker in the editor</source>
-        <translation type="unfinished"></translation>
+        <translation>Imprimir localização de depuração na Janela de Comandos, além do marcador no editor</translation>
     </message>
     <message>
         <location line="+16"/>
         <source>Command Window Colors</source>
-        <translation type="unfinished"></translation>
+        <translation>Cores da Janela de Comandos</translation>
     </message>
     <message>
         <location line="+172"/>
         <source>Show complete path in title</source>
         <translation>Exibir caminho completo no título</translation>
     </message>
     <message>
         <location line="+100"/>
@@ -4061,22 +4080,22 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+64"/>
         <source>Show completion list automatically</source>
         <translation>Exibir automaticamente lista de auto-completar</translation>
     </message>
     <message>
         <location line="+34"/>
         <source>Debugging</source>
-        <translation type="unfinished"></translation>
+        <translation>Depuração</translation>
     </message>
     <message>
         <location line="+8"/>
         <source>Always show debug breakpoints and pointers (opens related file if closed)</source>
-        <translation type="unfinished"></translation>
+        <translation>Sempre exibir apontadores e pontos de parada na depuração (abrundo arquivos relacionados se estiverem fechados)</translation>
     </message>
     <message>
         <location line="+15"/>
         <source>File handling</source>
         <translation>Manipulação de arquivos</translation>
     </message>
     <message>
         <location line="+19"/>
@@ -4121,17 +4140,17 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+22"/>
         <source>Restore last directory of previous session</source>
         <translation>Restaurar diretório da sessão anterior</translation>
     </message>
     <message>
         <location line="+32"/>
         <source>Initial file browser directory (only if not synchronized with initial working directory of Octave)</source>
-        <translation type="unfinished"></translation>
+        <translation>Diretório inicial do navegador de arquivos (somente se não estiver sincronizado com o diretório de trabalho inicial do Octave)</translation>
     </message>
     <message>
         <location line="+36"/>
         <source>Extensions of files to be opened in the default text editor (separated by &quot;;&quot;):</source>
         <translation>Extensões de arquivos a serem abertas no editor de texto padrão (separadas por &quot;;&quot;):</translation>
     </message>
     <message>
         <location line="+39"/>
@@ -4141,17 +4160,17 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+35"/>
         <source>Colors for variable attributes</source>
         <translation>Cores para atributos de variáveis</translation>
     </message>
     <message>
         <location line="+107"/>
         <source>Use Command Window font</source>
-        <translation type="unfinished"></translation>
+        <translation>Usar a fonte da Janela de Comandos</translation>
     </message>
     <message>
         <location line="+196"/>
         <source>Import shortcut set</source>
         <translation>Importar conjunto de teclas de atalho</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -4237,17 +4256,17 @@ Gostaria de salvar o conjunto atual de teclas de atalho ou cancelar a ação?</translation>
     <message>
         <location line="+12"/>
         <source>Use alternating row colors</source>
         <translation>Usar cores de linha alternadas</translation>
     </message>
     <message>
         <location line="+59"/>
         <source>Disable global shortcuts when Command Window has focus</source>
-        <translation type="unfinished"></translation>
+        <translation>Desativar teclas de atalho globais quando a Janela de Comando tiver foco</translation>
     </message>
     <message>
         <location line="+211"/>
         <source>HttpProxy</source>
         <translation>Proxy HTTP</translation>
     </message>
     <message>
         <location line="-2559"/>
diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -402,32 +402,34 @@ namespace octave
       query_string = queries.first ().wordList.join (" ");
     else
       query_string = "";
 #endif
 
     // Get quoted search strings first, then take first string as fall back
     QRegExp rx ("\"([^\"]*)\"");
     if (rx.indexIn (query_string, 0) != -1)
-      m_query_string = rx.cap (1);
+      m_internal_search = rx.cap (1);
     else
-      m_query_string = query_string.split (" ", QString::SkipEmptyParts).first ();
+      m_internal_search = query_string.split (" ", QString::SkipEmptyParts).first ();
 
     m_help_engine->searchEngine ()->search (queries);
   }
 
   void documentation::global_search_started (void)
   {
     qApp->setOverrideCursor(QCursor(Qt::WaitCursor));
   }
 
   void documentation::global_search_finished (int)
   {
     if (! m_internal_search.isEmpty ())
       {
+        m_query_string = m_internal_search;
+
         QHelpSearchEngine *search_engine = m_help_engine->searchEngine ();
         if (search_engine)
           {
 #if defined (HAVE_QHELPSEARCHQUERYWIDGET_SEARCHINPUT)
             QVector<QHelpSearchResult> res
               = search_engine->searchResults (0, search_engine->searchResultCount ());
 #else
             QList< QPair<QString, QString> > res
@@ -468,19 +470,19 @@ namespace octave
                             else if (url.isEmpty ())
                               url = tmpurl;
                           }
                       }
                   }
 
                 if (! url.isEmpty ())
                   {
-                    connect (this, SIGNAL (show_single_result (const QUrl)),
-                             m_doc_browser,
-                             SLOT (handle_index_clicked (const QUrl)));
+                    connect (this, SIGNAL (show_single_result (const QUrl&)),
+                             this,
+                             SLOT (handle_search_result_clicked (const QUrl&)));
 
                     emit show_single_result (url);
                   }
               }
            }
 
         m_internal_search = QString ();
       }
@@ -495,22 +497,28 @@ namespace octave
 
     // Select all occurrences of matching text
     select_all_occurrences (m_query_string);
 
     // Open search widget with matching text as search string
     m_find_line_edit->setText (m_query_string);
     m_find_line_edit->parentWidget ()->show ();
 
-    // Go to to first occurrence of search text. Going to the end and then
-    // search backwards until the last occurrence ensures the search text
-    // is visible in the first line of the visible part of the text.
-    m_doc_browser->moveCursor (QTextCursor::End);
-    while (m_doc_browser->find (m_find_line_edit->text (),
-                                QTextDocument::FindBackward));
+    // If no occurrence can be found go to the top of the page
+    if (! m_doc_browser->find (m_find_line_edit->text ()))
+      m_doc_browser->moveCursor (QTextCursor::Start);
+    else
+      {
+        // Go to to first occurrence of search text. Going to the end and then
+        // search backwards until the last occurrence ensures the search text
+        // is visible in the first line of the visible part of the text.
+        m_doc_browser->moveCursor (QTextCursor::End);
+        while (m_doc_browser->find (m_find_line_edit->text (),
+                                    QTextDocument::FindBackward));
+      }
   }
 
   void documentation::select_all_occurrences (const QString& text)
   {
     // Get highlight background and text color
     QPalette pal = QApplication::palette ();
     QTextCharFormat format;
     QColor col = pal.color (QPalette::Highlight);
diff --git a/libgui/src/documentation.h b/libgui/src/documentation.h
--- a/libgui/src/documentation.h
+++ b/libgui/src/documentation.h
@@ -122,17 +122,17 @@ namespace octave
     void handle_cursor_position_change (void);
     void handle_search_result_clicked (const QUrl& url);
 
     void update_history_menus (void);
     void open_hist_url (QAction *a);
 
   signals:
 
-    void show_single_result (const QUrl);
+    void show_single_result (const QUrl&);
 
   private:
 
     void construct_tool_bar (void);
     QAction *add_action (const QIcon& icon, const QString& text,
                          const char *member, QWidget *receiver = nullptr,
                          QToolBar *tool_bar = nullptr);
     void update_history (int new_count, QAction **actions);
diff --git a/libinterp/corefcn/ls-hdf5.cc b/libinterp/corefcn/ls-hdf5.cc
--- a/libinterp/corefcn/ls-hdf5.cc
+++ b/libinterp/corefcn/ls-hdf5.cc
@@ -1003,17 +1003,17 @@ save_hdf5_empty (octave_hdf5_id loc_id, 
     }
 
   retval = H5Dwrite (data_hid, H5T_NATIVE_IDX, octave_H5S_ALL, octave_H5S_ALL,
                      octave_H5P_DEFAULT, dims) >= 0;
 
   H5Dclose (data_hid);
   H5Sclose (space_hid);
 
-  if (retval >= 0)
+  if (retval)
     retval = hdf5_add_attr (loc_id, "OCTAVE_EMPTY_MATRIX");
 
   return (retval == 0 ? 1 : retval);
 
 #else
   err_disabled_feature ("save_hdf5_empty", "HDF5");
 #endif
 }
diff --git a/libinterp/corefcn/oct-stream.cc b/libinterp/corefcn/oct-stream.cc
--- a/libinterp/corefcn/oct-stream.cc
+++ b/libinterp/corefcn/oct-stream.cc
@@ -353,17 +353,17 @@ namespace octave
     void add_elt_to_list (int width, bool discard, char type, char modifier,
                           const std::string& char_class = "");
 
     void process_conversion (const std::string& s, size_t& i, size_t n,
                              int& width, bool& discard, char& type,
                              char& modifier);
 
     int finish_conversion (const std::string& s, size_t& i, size_t n,
-                           int& width, bool discard, char& type,
+                           int width, bool discard, char& type,
                            char modifier);
   };
 
   scanf_format_list::scanf_format_list (const std::string& s)
     : nconv (0), curr_idx (0), fmt_elts (), buf ()
   {
     size_t n = s.length ();
 
@@ -564,17 +564,17 @@ namespace octave
           break;
       }
 
     nconv = -1;
   }
 
   int
   scanf_format_list::finish_conversion (const std::string& s, size_t& i,
-                                        size_t n, int& width, bool discard,
+                                        size_t n, int width, bool discard,
                                         char& type, char modifier)
   {
     int retval = 0;
 
     std::string char_class;
 
     size_t beg_idx = std::string::npos;
     size_t end_idx = std::string::npos;
@@ -885,17 +885,17 @@ namespace octave
 
             switch (s[i])
               {
               case '%':
                 {
                   if (empty_buf)
                     {
                       process_conversion (s, i, n, args, flags, fw, prec,
-                                          type, modifier);
+                                          modifier, type);
 
                       // If there is nothing in the buffer, then
                       // add_elt_to_list must have just been called, so we
                       // are already done with the current element and we
                       // don't need to call add_elt_to_list if this is our
                       // last trip through the loop.
 
                       have_more = (buf.tellp () != 0);
@@ -1751,17 +1751,17 @@ namespace octave
                           char type,
                           const std::string& char_class = std::string ());
 
     void process_conversion (const std::string& s, size_t& i, size_t n);
 
     std::string parse_char_class (const std::string& pattern) const;
 
     int finish_conversion (const std::string& s, size_t& i, size_t n,
-                           unsigned int& width, int& prec, int& bitwidth,
+                           unsigned int width, int prec, int bitwidth,
                            octave_value& val_type,
                            bool discard, char& type);
   };
 
   // Main class to implement textscan.  Read data and parse it
   // according to a format.
   //
   // The calling sequence is
@@ -2328,18 +2328,18 @@ namespace octave
 
     retval.resize (out);
 
     return retval;
   }
 
   int
   textscan_format_list::finish_conversion (const std::string& s, size_t& i,
-                                           size_t n, unsigned int& width,
-                                           int& prec, int& bitwidth,
+                                           size_t n, unsigned int width,
+                                           int prec, int bitwidth,
                                            octave_value& val_type, bool discard,
                                            char& type)
   {
     int retval = 0;
 
     std::string char_class;
 
     size_t beg_idx = std::string::npos;
@@ -2452,17 +2452,16 @@ namespace octave
     int retval = 0;
 
     // read line, creating output_container as we go
     while (! ds.eof ())
       {
         bool already_skipped_delim = false;
         ts.skip_whitespace (ds);
         ds.progress_benchmark ();
-        bool progress = false;
         ts.scan_complex (ds, *fmt_elts[0], val);
         if (ds.fail ())
           {
             ds.clear (ds.rdstate () & ~std::ios::failbit);
 
             if (ds.eof ())
               break;
 
@@ -2487,17 +2486,17 @@ namespace octave
         else
           val_type = octave_value (ComplexNDArray (dv, val));
 
         output_container.push_back (val_type);
 
         if (! already_skipped_delim)
           ts.skip_delim (ds);
 
-        if (! progress && ds.no_progress ())
+        if (ds.no_progress ())
           break;
 
         nconv++;
       }
 
     output_container.pop_front (); // discard empty element from constructor
 
     // Create fmt_list now that the size is known
@@ -2638,17 +2637,17 @@ namespace octave
 
     // Read the data.  This is the main loop.
     if (! err)
       {
         for (/* row set ~30 lines above */; row < ntimes || ntimes == -1; row++)
           {
             if (row == 0 || row >= size)
               {
-                size += size+1;
+                size += (size+1);
                 for (auto& col : out)
                   col = col.resize (dim_vector (size, 1), 0);
               }
 
             row_idx(0) = row;
             err = read_format_once (is, fmt_list, out, row_idx, done_after);
 
             if ((err & ~1) > 0 || ! is || (lines >= ntimes && ntimes > -1))
@@ -2781,18 +2780,17 @@ namespace octave
 
     // Read fractional part, up to specified precision
     if (ch == '.' && width_left)
       {
         double multiplier = 1;
         int precision = fmt.prec;
         int i;
 
-        if (width_left)
-          width_left--;                // Consider width of '.'
+        width_left--;                  // Consider width of '.'
 
         if (precision == -1)
           precision = 1<<30;           // FIXME: Should be MAXINT
 
         if (! valid)                   // if there was nothing before '.'...
           is.get ();                   // ...ch was a "peek", not "get".
 
         for (i = 0; i < precision; i++)
@@ -2830,26 +2828,24 @@ namespace octave
         if (ch1 == '-' || ch1 == '+' || (ch1 >= '0' && ch1 <= '9'))
           {
             // if 1.0e+$ or some such, this will set failbit, as we want
             width_left--;                         // count "E"
             int exp = 0;
             int exp_sign = 1;
             if (ch1 == '+')
               {
-                if (width_left)
-                  width_left--;
+                width_left--;
                 is.get ();
               }
             else if (ch1 == '-')
               {
+                width_left--;
                 exp_sign = -1;
                 is.get ();
-                if (width_left)
-                  width_left--;
               }
             valid = false;
             while (width_left-- && is && (ch = is.get ()) >= '0' && ch <= '9')
               {
                 exp = exp*10 + ch - '0';
                 valid = true;
               }
             width_left++;
diff --git a/libinterp/corefcn/pr-output.cc b/libinterp/corefcn/pr-output.cc
--- a/libinterp/corefcn/pr-output.cc
+++ b/libinterp/corefcn/pr-output.cc
@@ -407,18 +407,16 @@ make_real_format (int digits, bool inf_o
         }
       else
         {
           ld = 1;
           rd = (prec > digits ? prec - digits : prec);
         }
 
       fw = 1 + ld + 1 + rd;
-      if (inf_or_nan && fw < 4)
-        fw = 4;
     }
 
   if (! (rat_format || bank_format || hex_format || bit_format)
       && (print_e || print_g || print_eng
           || ld + rd > pr_output_traits<T>::digits10
           || fw > pr_output_traits<T>::max_field_width))
     {
       if (print_g)
@@ -509,18 +507,16 @@ make_real_matrix_format (int x_max, int 
     {
       fw = 9;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = (x_max > x_min ? x_max : x_min);
       fw = (digits <= 0 ? 5 : digits + 4);
-      if (inf_or_nan && fw < 5)
-        fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       fw = 2 * sizeof (T);
       rd = 0;
     }
   else if (bit_format)
@@ -682,18 +678,16 @@ make_complex_format (int x_max, int x_mi
       r_fw = 0;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = r_x;
       i_fw = 0;
       r_fw = (digits <= 0 ? 5 : digits + 4);
-      if (inf_or_nan && r_fw < 5)
-        r_fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       r_fw = 2 * sizeof (T);
       i_fw = 2 * sizeof (T);
       rd = 0;
     }
@@ -750,21 +744,16 @@ make_complex_format (int x_max, int x_mi
           x_min = -x_min + 1;
         }
 
       ld = (ld_max > ld_min ? ld_max : ld_min);
       rd = (rd_max > rd_min ? rd_max : rd_min);
 
       i_fw = ld + 1 + rd;
       r_fw = i_fw + 1;
-      if (inf_or_nan && i_fw < 3)
-        {
-          i_fw = 3;
-          r_fw = 4;
-        }
     }
 
   if (! (rat_format || bank_format || hex_format || bit_format)
       && (print_e || print_eng || print_g
           || ld + rd > pr_output_traits<T>::digits10
           || r_fw > pr_output_traits<T>::max_field_width
           || i_fw > pr_output_traits<T>::max_field_width))
     {
@@ -901,18 +890,16 @@ make_complex_matrix_format (int x_max, i
       r_fw = 9;
       rd = 0;
     }
   else if (bank_format)
     {
       int digits = (r_x_max > r_x_min ? r_x_max : r_x_min);
       i_fw = 0;
       r_fw = (digits <= 0 ? 5 : digits + 4);
-      if (inf_or_nan && r_fw < 5)
-        r_fw = 5;
       rd = 2;
     }
   else if (hex_format)
     {
       r_fw = 2 * sizeof (T);
       i_fw = 2 * sizeof (T);
       rd = 0;
     }
@@ -1832,17 +1819,17 @@ octave_print_matrix_internal (std::ostre
                   for (octave_idx_type j = col; j < lim; j++)
                     {
                       octave_quit ();
 
                       if (i == 0 && j == 0)
                         os << "[ ";
                       else
                         {
-                          if (j > col && j < lim)
+                          if (j > col)
                             os << ", ";
                           else
                             os << "  ";
                         }
 
                       pr_float (os, fmt, m(i,j));
                     }
 
@@ -1948,17 +1935,17 @@ octave_print_diag_matrix_internal (std::
               for (octave_idx_type j = col; j < lim; j++)
                 {
                   octave_quit ();
 
                   if (j == 0)
                     os << "[ ";
                   else
                     {
-                      if (j > col && j < lim)
+                      if (j > col)
                         os << ", ";
                       else
                         os << "  ";
                     }
 
                   pr_float (os, fmt, m(j,j));
                 }
 
@@ -2086,18 +2073,17 @@ void print_nd_array (std::ostream& os, c
               page.print_with_name (os, nm);
             }
           else
             {
               page.print_name_tag (os, nm);
               page.print_raw (os);
             }
 
-          if (i < m)
-            NDA_T::increment_index (ra_idx, dims, 2);
+          NDA_T::increment_index (ra_idx, dims, 2);
         }
     }
 }
 
 void
 octave_print_internal (std::ostream& os, const NDArray& nda,
                        bool pr_as_read_syntax, int extra_indent)
 {
@@ -2242,17 +2228,17 @@ octave_print_internal (std::ostream& os,
               for (octave_idx_type j = col; j < lim; j++)
                 {
                   octave_quit ();
 
                   if (j == 0)
                     os << "[ ";
                   else
                     {
-                      if (j > col && j < lim)
+                      if (j > col)
                         os << ", ";
                       else
                         os << "  ";
                     }
 
                   os << pvec (j);
                 }
 
@@ -2686,18 +2672,17 @@ octave_print_internal (std::ostream& os,
                 os << "  " << page(ii,jj);
 
               os << "\n";
             }
 
           if (i < m - 1)
             os << "\n";
 
-          if (i < m)
-            increment_index (ra_idx, dims, 2);
+          increment_index (ra_idx, dims, 2);
         }
     }
 }
 
 template <typename T>
 class
 octave_print_conv
 {
diff --git a/libinterp/dldfcn/__eigs__.cc b/libinterp/dldfcn/__eigs__.cc
--- a/libinterp/dldfcn/__eigs__.cc
+++ b/libinterp/dldfcn/__eigs__.cc
@@ -236,17 +236,17 @@ Undocumented internal function.
           if (args(0).issparse ())
             {
               ascm = (args(0).sparse_complex_matrix_value ());
               a_is_sparse = true;
             }
           else
             acm = (args(0).complex_matrix_value ());
           a_is_complex = true;
-          symmetric = false; // ARPACK doesn't special case complex symmetric
+          symmetric = false;  // ARPACK doesn't special case complex symmetric
           sym_tested = true;
         }
       else
         {
           if (args(0).issparse ())
             {
               asmm = (args(0).sparse_matrix_value ());
               a_is_sparse = true;
@@ -322,27 +322,41 @@ Undocumented internal function.
       if (! args(3+arg_offset).isstruct ())
         error ("eigs: OPTS argument must be a structure");
 
       octave_scalar_map map = args(3+arg_offset).xscalar_map_value ("eigs: OPTS argument must be a scalar structure");
 
       octave_value tmp;
 
       // issym is ignored for complex matrix inputs
-      tmp = map.getfield ("issym");
-      if (tmp.is_defined () && ! sym_tested)
+      if (! sym_tested)
         {
-          symmetric = tmp.double_value () != 0.0;
-          sym_tested = true;
+          tmp = map.getfield ("issym");
+          if (tmp.is_defined ())
+            {
+              if (tmp.numel () != 1)
+                error ("eigs: OPTS.issym must be a scalar value");
+
+              symmetric = tmp.xbool_value ("eigs: OPTS.issym must be a logical value");
+              sym_tested = true;
+            }
         }
 
       // isreal is ignored if A is not a function
-      tmp = map.getfield ("isreal");
-      if (tmp.is_defined () && have_a_fun)
-        a_is_complex = ! (tmp.double_value () != 0.0);
+      if (have_a_fun)
+        {
+          tmp = map.getfield ("isreal");
+          if (tmp.is_defined ())
+            {
+              if (tmp.numel () != 1)
+                error ("eigs: OPTS.isreal must be a scalar value");
+
+              a_is_complex = ! tmp.xbool_value ("eigs: OPTS.isreal must be a logical value");
+            }
+        }
 
       tmp = map.getfield ("tol");
       if (tmp.is_defined ())
         tol = tmp.double_value ();
 
       tmp = map.getfield ("maxit");
       if (tmp.is_defined ())
         maxit = tmp.nint_value ();
@@ -361,17 +375,22 @@ Undocumented internal function.
         }
 
       tmp = map.getfield ("disp");
       if (tmp.is_defined ())
         disp = tmp.nint_value ();
 
       tmp = map.getfield ("cholB");
       if (tmp.is_defined ())
-        cholB = tmp.double_value () != 0.0;
+        {
+          if (tmp.numel () != 1)
+            error ("eigs: OPTS.cholB must be a scalar value");
+
+          cholB = tmp.xbool_value ("eigs: OPTS.cholB must be a logical value");
+        }
 
       tmp = map.getfield ("permB");
       if (tmp.is_defined ())
         permB = ColumnVector (tmp.vector_value ()) - 1.0;
     }
 
   if (nargin > (4+arg_offset))
     error ("eigs: incorrect number of arguments");
diff --git a/libinterp/parse-tree/pt-assign.cc b/libinterp/parse-tree/pt-assign.cc
--- a/libinterp/parse-tree/pt-assign.cc
+++ b/libinterp/parse-tree/pt-assign.cc
@@ -35,17 +35,17 @@ namespace octave
 {
   // Simple assignment expressions.
 
   tree_simple_assignment::tree_simple_assignment (tree_expression *le,
                                                   tree_expression *re,
                                                   bool plhs, int l, int c,
                                                   octave_value::assign_op t)
     : tree_expression (l, c), m_lhs (le), m_rhs (re), m_preserve (plhs),
-      m_etype (t)
+      m_ans_assign (), m_etype (t)
   { }
 
   tree_simple_assignment::~tree_simple_assignment (void)
   {
     if (! m_preserve)
       delete m_lhs;
 
     delete m_rhs;
diff --git a/libinterp/parse-tree/pt-eval.cc b/libinterp/parse-tree/pt-eval.cc
--- a/libinterp/parse-tree/pt-eval.cc
+++ b/libinterp/parse-tree/pt-eval.cc
@@ -185,16 +185,19 @@ namespace octave
     if (retval == EOF)
       {
         if (interactive)
           octave_stdout << "\n";
 
         retval = 0;
       }
 
+    // Clean up memory
+    delete lxr;
+
     return retval;
   }
 
   octave_value_list
   tree_evaluator::eval_string (const std::string& eval_str, bool silent,
                                int& parse_status, int nargout)
   {
     octave_value_list retval;
diff --git a/liboctave/array/CDiagMatrix.cc b/liboctave/array/CDiagMatrix.cc
--- a/liboctave/array/CDiagMatrix.cc
+++ b/liboctave/array/CDiagMatrix.cc
@@ -242,29 +242,29 @@ ComplexRowVector
 ComplexDiagMatrix::row (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= r)
     (*current_liboctave_error_handler) ("invalid row selection");
 
   ComplexRowVector retval (c, 0.0);
-  if (r <= c || (r > c && i < c))
+  if (r <= c || i < c)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 ComplexRowVector
 ComplexDiagMatrix::row (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid row selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return row (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return row (rows () - 1);
   else
     (*current_liboctave_error_handler) ("invalid row selection");
 }
 
@@ -272,29 +272,29 @@ ComplexColumnVector
 ComplexDiagMatrix::column (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= c)
     (*current_liboctave_error_handler) ("invalid column selection");
 
   ComplexColumnVector retval (r, 0.0);
-  if (r >= c || (r < c && i < r))
+  if (r >= c || i < r)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 ComplexColumnVector
 ComplexDiagMatrix::column (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid column selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return column (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return column (cols () - 1);
   else
     (*current_liboctave_error_handler) ("invalid column selection");
 }
 
diff --git a/liboctave/array/dDiagMatrix.cc b/liboctave/array/dDiagMatrix.cc
--- a/liboctave/array/dDiagMatrix.cc
+++ b/liboctave/array/dDiagMatrix.cc
@@ -165,29 +165,29 @@ RowVector
 DiagMatrix::row (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= r)
     (*current_liboctave_error_handler) ("invalid row selection");
 
   RowVector retval (c, 0.0);
-  if (r <= c || (r > c && i < c))
+  if (r <= c || i < c)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 RowVector
 DiagMatrix::row (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid row selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return row (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return row (rows () - 1);
   else
     (*current_liboctave_error_handler) ("invalid row selection");
 }
 
@@ -195,29 +195,29 @@ ColumnVector
 DiagMatrix::column (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= c)
     (*current_liboctave_error_handler) ("invalid column selection");
 
   ColumnVector retval (r, 0.0);
-  if (r >= c || (r < c && i < r))
+  if (r >= c || i < r)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 ColumnVector
 DiagMatrix::column (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid column selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return column (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return column (cols () - 1);
   else
     (*current_liboctave_error_handler) ("invalid column selection");
 }
 
diff --git a/liboctave/array/fCDiagMatrix.cc b/liboctave/array/fCDiagMatrix.cc
--- a/liboctave/array/fCDiagMatrix.cc
+++ b/liboctave/array/fCDiagMatrix.cc
@@ -246,29 +246,29 @@ FloatComplexRowVector
 FloatComplexDiagMatrix::row (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= r)
     (*current_liboctave_error_handler) ("invalid row selection");
 
   FloatComplexRowVector retval (c, 0.0);
-  if (r <= c || (r > c && i < c))
+  if (r <= c || i < c)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 FloatComplexRowVector
 FloatComplexDiagMatrix::row (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid row selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return row (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return row (rows () - 1);
   else
     (*current_liboctave_error_handler) ("invalid row selection");
 }
 
@@ -276,29 +276,29 @@ FloatComplexColumnVector
 FloatComplexDiagMatrix::column (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= c)
     (*current_liboctave_error_handler) ("invalid column selection");
 
   FloatComplexColumnVector retval (r, 0.0);
-  if (r >= c || (r < c && i < r))
+  if (r >= c || i < r)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 FloatComplexColumnVector
 FloatComplexDiagMatrix::column (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid column selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return column (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return column (cols () - 1);
   else
     (*current_liboctave_error_handler) ("invalid column selection");
 }
 
diff --git a/liboctave/array/fDiagMatrix.cc b/liboctave/array/fDiagMatrix.cc
--- a/liboctave/array/fDiagMatrix.cc
+++ b/liboctave/array/fDiagMatrix.cc
@@ -165,29 +165,29 @@ FloatRowVector
 FloatDiagMatrix::row (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= r)
     (*current_liboctave_error_handler) ("invalid row selection");
 
   FloatRowVector retval (c, 0.0);
-  if (r <= c || (r > c && i < c))
+  if (r <= c || i < c)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 FloatRowVector
 FloatDiagMatrix::row (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid row selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return row (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return row (rows () - 1);
   else
     (*current_liboctave_error_handler) ("invalid row selection");
 }
 
@@ -195,29 +195,29 @@ FloatColumnVector
 FloatDiagMatrix::column (octave_idx_type i) const
 {
   octave_idx_type r = rows ();
   octave_idx_type c = cols ();
   if (i < 0 || i >= c)
     (*current_liboctave_error_handler) ("invalid column selection");
 
   FloatColumnVector retval (r, 0.0);
-  if (r >= c || (r < c && i < r))
+  if (r >= c || i < r)
     retval.elem (i) = elem (i, i);
 
   return retval;
 }
 
 FloatColumnVector
 FloatDiagMatrix::column (char *s) const
 {
   if (! s)
     (*current_liboctave_error_handler) ("invalid column selection");
 
-  char c = *s;
+  char c = s[0];
   if (c == 'f' || c == 'F')
     return column (static_cast<octave_idx_type> (0));
   else if (c == 'l' || c == 'L')
     return column (cols () - 1);
   else
     (*current_liboctave_error_handler) ("invalid column selection");
 }
 
diff --git a/liboctave/system/lo-sysdep.cc b/liboctave/system/lo-sysdep.cc
--- a/liboctave/system/lo-sysdep.cc
+++ b/liboctave/system/lo-sysdep.cc
@@ -165,17 +165,18 @@ namespace octave
       int new_len = name.length () + value.length () + 2;
 
       // FIXME: This leaks memory, but so would a call to setenv.
       // Short of extreme measures to track memory, altering the environment
       // always leaks memory, but the saving grace is that the leaks are small.
 
       char *new_item = static_cast<char *> (std::malloc (new_len));
 
-      sprintf (new_item, "%s=%s", name.c_str (), value.c_str ());
+      if (new_item)
+        sprintf (new_item, "%s=%s", name.c_str (), value.c_str ());
 
       // As far as I can see there's no way to distinguish between the
       // various errors; putenv doesn't have errno values.
 
 #if defined (OCTAVE_USE_WINDOWS_API)
       wchar_t *wnew_item = u8_to_wchar (new_item);
       std::free (static_cast<void *> (new_item));
       if (_wputenv (wnew_item) < 0)
diff --git a/liboctave/system/lo-sysinfo.cc b/liboctave/system/lo-sysinfo.cc
--- a/liboctave/system/lo-sysinfo.cc
+++ b/liboctave/system/lo-sysinfo.cc
@@ -61,20 +61,17 @@ namespace octave
           int v_minor = 0;
           int v_patch = 0;
           flexi_f_ptr (&v_major, &v_minor, &v_patch);
 
           std::ostringstream s;
           s << "FlexiBLAS Version "
             << v_major << "." << v_minor << "." << v_patch;
 
-          if (! retval.empty ())
-            retval += "\n";
-
-          retval += s.str ();
+          retval = s.str ();
         }
 
       // OpenBLAS
       typedef char * (*open_fcn_type) (void);
       open_fcn_type open_f_ptr = reinterpret_cast<open_fcn_type>
                                  (dyn_libs.search ("openblas_get_config"));
 
       if (open_f_ptr)
diff --git a/liboctave/util/oct-rl-edit.c b/liboctave/util/oct-rl-edit.c
--- a/liboctave/util/oct-rl-edit.c
+++ b/liboctave/util/oct-rl-edit.c
@@ -39,17 +39,18 @@ along with Octave; see the file COPYING.
   if (ss) \
     { \
       free (ss); \
       ss = 0; \
     } \
  \
   ss = malloc (strlen (s) + 1); \
  \
-  strcpy (ss, s)
+  if (ss) \
+    strcpy (ss, s)
 
 void
 octave_rl_redisplay (void)
 {
   rl_redisplay ();
 }
 
 int
diff --git a/scripts/sparse/eigs.m b/scripts/sparse/eigs.m
--- a/scripts/sparse/eigs.m
+++ b/scripts/sparse/eigs.m
@@ -866,26 +866,26 @@ endfunction
 %! d1diag = diag (d1);
 %! [d1diag, idx] = sort (d1diag);
 %! v1 = v1(:, idx);
 %! assert (abs (v), abs (R \ v1), 1e-12);
 %!testif HAVE_ARPACK, HAVE_UMFPACK
 %! A = toeplitz (sparse (1:10));
 %! B = toeplitz (sparse ([1, 1], [1, 2], [2, 1], 1, 10));
 %! R = chol (B);
-%! opts.cholB = R;
+%! opts.cholB = true;
 %! [v, d] = eigs (A, R, 4, "lm", opts);
 %! for i = 1:4
 %!   assert (A * v(:,i), d(i, i) * B * v(:,i), 1e-12)
 %! endfor
 %!testif HAVE_ARPACK, HAVE_UMFPACK
 %! A = toeplitz (sparse (1:10));
 %! B = toeplitz (sparse ([1, 1], [1, 2], [2, 1], 1, 10));
 %! [R, ~, permB] = chol (B, "vector");
-%! opts.cholB = R;
+%! opts.cholB = true;
 %! opts.permB = permB;
 %! [v, d] = eigs (A, R, 4, "lm", opts);
 %! for i = 1:4
 %!   assert (A * v(:,i), d(i, i) * B * v(:,i), 1e-12)
 %! endfor
 
 
 #### FULL MATRIX VERSIONS ####
